### PR TITLE
add: reservation unit edit seasonal times

### PR DESCRIPTION
**Pull request by joonatank**

### CODE REVIEW

File: apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+  grid-template-columns: repeat(2, auto);
+  /* TODO 1fr for both the error texts creates a problem where one error is visible it scales all the boxes
+   * creating extra white space where the error could be (but isn't) */
+  grid-template-rows:
+    [day] auto
+    [a0] auto auto auto
+    [a1] auto auto auto
+    [controls] auto
+    [controls] auto;
```
Comment (joonatank) [inline apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx:35]: combine `[controls] auto auto`

File: apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+    "a0 a0"
+    "a1 a1"
+    "a1 a1"
+    "a1 a1"
+    "controls controls"
+    "controls controls";
+
+  grid-template-columns: repeat(2, auto);
+  /* TODO 1fr for both the error texts creates a problem where one error is visible it scales all the boxes
+   * creating extra white space where the error could be (but isn't) */
```
Comment (joonatank) [inline apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx:29]: REMOVE (fixed already)

File: apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+  align-items: end;
+  display: grid;
+  grid: subgrid / subgrid;
+  grid-area: a;
+
+  /* use more margin instead of gap because we have invisible columns (mobile is vertical layout) */
+  margin-bottom: var(--spacing-xs);
+  gap: var(--spacing-2-xs);
+
+  /* using area in the smaller ones; TODO this should also use it */
```
Comment (joonatank) [inline apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx:81]: REMOVE

File: apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+`;
+
+const StyledTimeInput = styled(TimeInput)`
+  & {
+    display: grid;
+    grid: subgrid / subgrid;
+    grid-row-end: span 3;
+    /* TODO should take as much space as possible because of the error message
+     * can't use the grid layout because that fixes too many things (unless we use px units)
+     * */
```
Comment (joonatank) [inline apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx:96]: REMOVE / REWORK

File: apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+    & > input {
+      min-width: 11ch;
+    }
+  }
+`;
+
+const gridSpan = css`
+  grid-row-start: 2;
```
Comment (joonatank) [inline apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx:104]: comment: line with the input field (line 2 in the grid)

File: apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+  align-self: end;
+`;
+const GridSpan = styled.span`
```
Comment (joonatank) [inline apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx:107]: AndSpan (not GridSpan)

File: apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+                      t
+                    )}
+                  />
+                )}
+              />
+            </SeasonalTimeWrapper>
+          </>
+        );
+      })}
+      {/* TODO move the css to styled components, otherwise it works */}
```
Comment (joonatank) [inline apps/admin-ui/src/spa/ReservationUnit/edit/SeasonalSection.tsx:245]: REMOVE

**Review by vincit-matu — APPROVED** [review]
We went through the possible pain points together, but LGTM.

**Pull request merged**